### DISCUSSION TITLE
M3 apple silicon

### DISCUSSION THREAD

**Discussion by tonyageall** [facebookresearch/esm#646] [General]
Hi, 

Calling on this workflow: 

# Extract per-residue representations (on CPU)
with torch.no_grad():
    results = model(batch_tokens, repr_layers=[33], return_contacts=True)
token_representations = results["representations"][33]

Is there a way to accelerate this on apples M3 silicon chip? 

Thanks

**Comment by oliverfleetwood** [comment]
See if you have MPS available. 

```python
if torch.backends.mps.is_available():  # for Macs
        device = torch.device("mps")
        print(f"MPS is available. using device #{str(device)}")
    else:
        print("CUDA or MPS device not available. Using CPU.")
        device = 'cpu'

results = model.to(device)(batch_tokens.to(device), repr_layers=[33], return_contacts=True)

```
